src/dune_lang/package_name.ml

```diff
@@ -49,11 +49,24 @@ module Opam_compatible = struct
 
     let of_string_opt s = Option.some_if (is_valid_string s) s
 
-    let make_valid s =
+    let escape s =
       let replaced = String.map s ~f:(fun c -> if is_valid_char c then c else '_') in
       if is_valid_string replaced then replaced else "p" ^ replaced
     ;;
 
+    let make_valid s =
```

Comment:
As @Alizter suggests, I think this needs a separate `make_valid_for_dependency` (or whatever you want to call it) function.

Reply:
I believe using a simpler `Dependency.decode` which uses the old `make_valid` under the hood, as per my last commit, solves this. The old way of doing it is preserved.